p2p/chain.py

```diff
@@ -67,12 +72,12 @@ class BaseHeaderChainSyncer(BaseService, PeerPoolSubscriber):
 
     def __init__(self,
                  chain: AsyncChain,
-                 chaindb: AsyncChainDB,
+                 db: 'AsyncHeaderDB',
                  peer_pool: PeerPool,
                  token: CancelToken = None) -> None:
         super().__init__(token)
         self.chain = chain
-        self.chaindb = chaindb
+        self.db = db
```

Comment:
This rename looks like it may cause confusion.  Our current naming conventions have been

```
- `db` for `BaseDB` type objects
- `chaindb` for `ChainDB` type objects
- `headerdb` for `HeaderDB` type objects.
```

If I saw this I'd assume that `db` was a `BaseDB` instance.